modules/apps/callbacks/fee_transfer_test.go

```diff
@@ -0,0 +1,197 @@
+		transferMemo string
+		expCallback  types.CallbackType
+		expSuccess   bool
+	}{
+		{
+			"success: transfer with no memo",
+			"",
+			"none",
+			true,
+		},
+		{
+			"success: dest callback",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true, // timeouts don't reach destination chain execution
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTypeTimeoutPacket,
+			true,
+		},
+		{
+			"success: dest callback with low gas (panic)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			"none", // timeouts don't reach destination chain execution
+			false,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			types.CallbackTypeTimeoutPacket,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			s.SetupFeeTransferTest()
+
+			fee := feetypes.NewFee(defaultRecvFee, defaultAckFee, defaultTimeoutFee)
+
+			s.ExecutePayPacketFeeMsg(fee)
+			preRelaySenderBalance := sdk.NewCoins(s.chainA.GetSimApp().BankKeeper.GetBalance(s.chainA.GetContext(), s.chainA.SenderAccount.GetAddress(), ibctesting.TestCoin.Denom))
+			s.ExecuteTransferTimeout(tc.transferMemo, 1)
+
+			// after incentivizing the packets
+			s.AssertHasExecutedExpectedCallbackWithFee(tc.expCallback, tc.expSuccess, true, preRelaySenderBalance, fee)
+		})
+	}
+}
+
+func (s *CallbacksTestSuite) ExecutePayPacketFeeMsg(fee feetypes.Fee) {
+	msg := feetypes.NewMsgPayPacketFee(
+		fee, s.path.EndpointA.ChannelConfig.PortID, s.path.EndpointA.ChannelID,
+		s.chainA.SenderAccount.GetAddress().String(), nil,
+	)
+
+	// fetch the account balance before fees are escrowed and assert the difference below
+	preEscrowBalance := s.chainA.GetSimApp().BankKeeper.GetBalance(s.chainA.GetContext(), s.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+
+	res, err := s.chainA.SendMsgs(msg)
+	s.Require().NotNil(res)
+	s.Require().NoError(err) // message committed
+
+	postEscrowBalance := s.chainA.GetSimApp().BankKeeper.GetBalance(s.chainA.GetContext(), s.chainA.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+	s.Require().Equal(postEscrowBalance.AddAmount(fee.Total().AmountOf(sdk.DefaultBondDenom)), preEscrowBalance)
+
+	// register counterparty address on chainB
+	// relayerAddress is address of sender account on chainB, but we will use it on chainA
+	// to differentiate from the chainA.SenderAccount for checking successful relay payouts
+	relayerAddress := s.chainB.SenderAccount.GetAddress()
+
+	msgRegister := feetypes.NewMsgRegisterCounterpartyPayee(
+		s.path.EndpointB.ChannelConfig.PortID, s.path.EndpointB.ChannelID,
+		s.chainB.SenderAccount.GetAddress().String(), relayerAddress.String(),
+	)
+	_, err = s.chainB.SendMsgs(msgRegister)
+	s.Require().NoError(err) // message committed
```

Comment:
I think we could trim this function down a bunch. We're kind of testing fee behaviour is working correctly here by asserting balances etc, imo the callbacks module should just assume a correct and working implementation.

We could simplify this to route the two msgs together, something like this:

```suggestion
	msgPayPacketFee := feetypes.NewMsgPayPacketFee(
		fee, s.path.EndpointA.ChannelConfig.PortID, s.path.EndpointA.ChannelID,
		s.chainA.SenderAccount.GetAddress().String(), nil,
	)

	// register counterparty address on chainB
	// relayerAddress is address of sender account on chainB, but we will use it on chainA
	// to differentiate from the chainA.SenderAccount for checking successful relay payouts
	relayerAddress := s.chainB.SenderAccount.GetAddress()
	msgRegisterCounterpartyPayee := feetypes.NewMsgRegisterCounterpartyPayee(
		s.path.EndpointB.ChannelConfig.PortID, s.path.EndpointB.ChannelID,
		s.chainB.SenderAccount.GetAddress().String(), relayerAddress.String(),
	)
	
	res, err := s.chainA.SendMsgs(msgPayPacketFee, msgRegisterCounterpartyPayee)
	s.Require().NotNil(res)
	s.Require().NoError(err) // message committed
```

Reply:
I was just trying to ensure callbacks doesn't break paying of the fees. This is also acceptable. I guess I wanna see what others think